public/tabs/chat/chat_page_content.tsx

```diff
@@ -37,6 +39,8 @@ export const ChatPageContent: React.FC<ChatPageContentProps> = React.memo((props
   const loading = props.messagesLoading || chatState.llmResponding;
   const chatActions = useChatActions();
   const registry = getIncontextInsightRegistry();
+  const configSchema = getConfigSchema();
+  const showRegenerate = configSchema.chat.regenerateButton;
```

Comment:
If we do this on line 133, then no changes are required for `message_bubble.tsx` and `message_action.tsx`
```
showRegenerate={isLatestOutput && showRegenerate}
```